src/types/fieldExtractor.ts

```diff
@@ -161,13 +161,18 @@ const idStrategy: StrategyExtractor = (contract) => {
   return extractor;
 }
 
-export const resolveEthereumTrackId: Resolver = (nft, contract) => {
+export const resolveTrackId: Resolver = (nft, contract) => {
   const extractor = idStrategy(contract)
-  const trackId = slugify(extractor(nft));
-  if (!trackId) {
+  const id = slugify(extractor(nft));
+  if (!id) {
     throw new Error(`ID not extracted correctly for nft: ${nft.id}`);
   }
-  return ethereumTrackId(nft.contractAddress, trackId);
+
+  // prefer the collection override over nft.contract address.
+  // this helps us handle dedeuplicating solana nfts (where each nft has a different contract)
+  const address = contract.typeMetadata?.collection || nft.contractAddress;
```

Comment:
tracks will use the metaFactory address rather than the mint address in the ID to avoid duplicates